### PR TITLE
feat(monitors): #1368 added monitor table in project dashboard

### LINKED ISSUE #1368: Show monitors and latest ping status on the dashboard page
Dependant on #1360

### PR DESCRIPTION
**Pull request by webkhushboo**
closes #1368

### Notes

A summary of what was achieved in this PR

- [ ] UI: Created monitor table 
- [ ] UI: Show monitor table on project dashboard page

### CODE REVIEW

**Comment by eddiejaoude** [comment]
@webkhushboo  After some thought, I changed it from table to list, what do you think? Please review code when you get time 🤓 

Screenshot 2019-08-05 07 28 31

File: web/src/app/projects/monitor-summary/monitor-summary.component.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<mat-card>
+  <mat-card-title>Monitors</mat-card-title>
+  <mat-card-subtitle>{{ monitors.length }} monitors</mat-card-subtitle>
+  <mat-card-content fxLayout="row">
+    <mat-list dense fxFlex="50">
+      <h3 mat-subheader>Currently Valid</h3>
```
Comment (webkhushboo) [inline web/src/app/projects/monitor-summary/monitor-summary.component.html]: We can change currently valid to latest valid ping ?

File: web/src/app/projects/monitor-summary/monitor-summary.component.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      <p *ngIf="!filterBy(true).length">No valid monitors</p>
+      <mat-list-item *ngFor="let valid of filterBy(true)">
+        <mat-icon mat-list-icon [matBadge]="valid.latestPing.duration" matBadgeColor="success" *ngIf="valid.latestPing.isValid">done</mat-icon>
+        <h4 mat-line>{{ valid.latestPing.url }}</h4>
+        <p mat-line>{{ valid.latestPing.statusCode }} | {{ valid.latestPing.duration }}ms | {{ valid.latestPing.type }} | {{ valid.latestPing.createdOn.toDate() | timeAgo }}</p>
+      </mat-list-item>
+    </mat-list>
+    <mat-list dense fxFlex="50">
+      <h3 mat-subheader>Currently Invalid</h3>
```
Comment (webkhushboo) [inline web/src/app/projects/monitor-summary/monitor-summary.component.html]: We can change currently invalid to latest invalid ping ?

File: web/src/app/projects/monitor-summary/monitor-summary.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  selector: 'dashboard-monitor-summary',
+  templateUrl: './monitor-summary.component.html',
+  styleUrls: ['./monitor-summary.component.scss'],
+})
+export class MonitorSummaryComponent {
+
+  @Input()
+  public monitors: MonitorModel[];
+
+  public filterBy(isValid: boolean): MonitorModel[] {
```
Comment (webkhushboo) [inline web/src/app/projects/monitor-summary/monitor-summary.component.ts]: We can add the comments here

File: web/src/app/projects/projects.module.ts [hunk 4]
```diff
@@ -19,6 +20,8 @@ import { ViewProjectComponent } from './view/view.component';
   declarations: [
     CreateEditProjectComponent,
     RepositoryComponent,
-    ViewProjectComponent],
+    ViewProjectComponent,
+    MonitorSummaryComponent,
```
Comment (webkhushboo) [inline web/src/app/projects/projects.module.ts]: Alphabetical order is not correct

File: web/src/app/projects/monitor-summary/monitor-summary.component.ts [hunk 5]
```diff
@@ -1,15 +1,12 @@
...
 import { Component, Input } from '@angular/core';
-import { MatTableDataSource } from '@angular/material';
-import { ActivatedRoute } from '@angular/router';
 
-// Third party modules
-import { Subscription } from 'rxjs';
-
-// Dashboard hub model and services
-import { ProjectService } from '../../core/services/index.service';
-import { MonitorModel, ProjectModel } from '../../shared/models/index.model';
```
Comment (eddiejaoude) [inline web/src/app/projects/monitor-summary/monitor-summary.component.ts]: I forgot to remove these 🤦‍♂ - thank you 🤓

**Pull request merged**